Java/core/src/main/java/com/amazon/randomcutforest/store/PointStore.java

```diff
@@ -433,25 +431,7 @@ public boolean isDirectLocationMap() {
      * @return for internal shingling, returns the last seen shingle
      */
     public double[] getInternalShingle() {
-        return (internalShinglingEnabled) ? Arrays.copyOf(internalShingle, dimensions) : null;
-    }
-
-    /**
-     * used for serialization
-     * 
-     * @return the free indices as a stack 9can be implicitly defined)
-     */
-    public int[] getFreeIndexes() {
-        return indexManager.getFreeIndexes();
-    }
-
-    /**
-     * used for serialization
-     * 
-     * @return the pointer to the stack of free indices
-     */
-    public int getFreeIndexPointer() {
-        return indexManager.getFreeIndexPointer();
+        return (internalShinglingEnabled) ? copyShingle() : null;
```

Comment:
It seems like this should throw an exception instead of returning null. Would cause problems in other places?
```
checkState(internalShinglingEnabled, "internal shingling is not enabled");
return copyShingle();
```

Reply:
The mappers were using the get() --  but it may be a good change for the mappers to not write nulls. Changing. Will cause the addition of two of the mapper files.